pkg/platform/local/platform.go

```diff
@@ -712,3 +730,70 @@ func (p *Platform) deletePreviousContainers(createFunctionOptions *platform.Crea
 
 	return previousHTTPPort, nil
 }
+
+func (p *Platform) ValidateFunctionContainersHealthiness() {
+	namespaces, err := p.GetNamespaces()
+	if err != nil {
+		p.Logger.WarnWith("Could not get namespaces", "err", err.Error())
+		return
+	}
+	for _, namespace := range namespaces {
+
+		// get functions for that namespace
+		functions, err := p.GetFunctions(&platform.GetFunctionsOptions{
+			Namespace: namespace,
+		})
+		if err != nil {
+			p.Logger.WarnWith("Could not get functions",
+				"namespace", namespace,
+				"err", err.Error())
+			continue
+		}
+		// For each function, we will check if its container is healthy
+		// in case it is not healthy (or container is missing), update function status
+		// and mark its state to error
+		for _, function := range functions {
+			functionConfig := function.GetConfig()
+			functionState := function.GetStatus().State
+			functionName := functionConfig.Meta.Name
+			if functionState != functionconfig.FunctionStateReady {
+				p.Logger.DebugWith("Skipping function not in ready state",
+					"functionName",
+					functionName,
+					"functionState",
+					functionState)
+				continue
+			}
+
+			// get function container id
+			containerID := p.getContainerNameByCreateFunctionOptions(&platform.CreateFunctionOptions{
+				FunctionConfig: functionconfig.Config{
+					Meta: functionconfig.Meta{
+						Name:      functionName,
+						Namespace: namespace,
+					},
+				},
+			})
+
+			p.Logger.DebugWith("Checking function container healthiness",
```

Comment:
again - will spam the logs
it might be worth to aggregate all ready and healthy functions and all non ready functions and log this at the end of the healthCheck iteration